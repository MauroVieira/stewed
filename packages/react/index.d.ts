/**
 * Required to avoid errors when importing these types of files
 */

declare module "*.svg?url" {
  const content: string;
  export default content;
}

declare module "*.svg" {
  const content: React.FunctionComponent<React.SVGAttributes<SVGElement>>;
  export default content;
}

declare module "*.md" {
  const content: string;
  export default content;
}

declare module "*.scss" {
  const classes: { [key: string]: string };
  export default classes;
}
