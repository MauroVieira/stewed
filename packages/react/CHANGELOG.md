# @stewed/react

## 0.1.0

### Minor Changes

- a302a43: Update all packages

### Patch Changes

- Updated dependencies [a302a43]
  - @stewed/utilities@0.2.0
  - @stewed/hooks@0.1.0
  - @stewed/tokens@0.2.0
