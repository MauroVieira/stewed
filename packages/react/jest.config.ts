import { jestConfig } from "@harmonix-hub/testing";

module.exports = {
  ...jestConfig(),
  testEnvironment: "jsdom",
  moduleNameMapper: {
    "\\.(css|less|scss|sss|styl)$": "<rootDir>/node_modules/jest-css-modules",
  }
};
