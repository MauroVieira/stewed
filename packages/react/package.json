{
  "name": "@stewed/react",
  "version": "0.0.0",
  "main": "./dist/index.js",
  "module": "./dist/index.mjs",
  "types": "./dist/index.d.ts",
  "sideEffects": false,
  "license": "MIT",
  "files": [
    "dist/**"
  ],
  "scripts": {
    "dev": "npx vite",
    "build": "npx vite build",
    "lint": "TIMING=1 eslint \"src/**/*.ts*\"",
    "clean": "rm -rf .turbo && rm -rf node_modules && rm -rf dist"
  },
  "devDependencies": {
    "@stewed/tsconfig": "workspace:*",
    "@stewed/utils": "workspace:*",
    "@stewed/lint": "workspace:*",
    "@types/react": "^18.0.9",
    "@types/react-dom": "^18.0.4",
    "eslint": "^8.15.0",
    "react": "^18.1.0",
    "tsup": "^5.10.1",
    "typescript": "^4.5.3",
    "sass": "^1.58.0",
    "@vitejs/plugin-react": "^1.3.2",
    "vite": "^2.9.15"
  },
  "publishConfig": {
    "access": "public"
  }
}
