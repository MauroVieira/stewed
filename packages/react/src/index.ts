export * from "./accordion";
export * from "./alert";
export * from "./aspect-ratio";
export * from "./avatar";
export * from "./badge";
export * from "./box";
export * from "./button";
export * from "./card";
export * from "./carousel";
export * from "./checkbox";
export * from "./divider";
export * from "./list-box";
export * from "./progress";
export * from "./radio";
export * from "./tabs";
export * from "./tag";
export * from "./text-area";
export * from "./text-field";
export * from "./theme";
export * from "./theme";
export * from "./toggle";
export * from "./typography";

// Utilities
export * from "./types";
