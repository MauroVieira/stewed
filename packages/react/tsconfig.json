{
  "extends": "@harmonix-hub/tsconfig/react.json",
  "compilerOptions": {
    "outDir": "./dist",
  },
  "include": ["custom.d.ts", "index.ts", "./src/**/*", "./src/index.ts"],
  "exclude": ["dist", "src/**/stories", "src/**/__tests__", "node_modules"]
}
