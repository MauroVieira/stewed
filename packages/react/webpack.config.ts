import path from "path";
// Types
import type { Configuration } from "webpack";

const webpackConfig: Configuration = {
  mode: "production",
  entry: "./src/index.ts",
  output: {
    filename: "index.js",
    library: {
      type: "module",
    },
    path: path.resolve(__dirname, "dist"),
  },
  experiments: {
    outputModule: true,
  },
  optimization: {
    minimize: true,
  },
  module: {
    rules: [
      {
        test: /\.s[ac]ss$/i,
        use: [
          "style-loader",
          {
            loader: "css-loader",
            options: {
              modules: {
                auto: true,
                localIdentName: "[hash:base64]",
              },
              sourceMap: true,
            },
          },
          "sass-loader",
        ],
      },
      {
        test: /\.tsx?$/i,
        exclude: /node_modules/,
        use: {
          loader: "ts-loader",
          options: {
            compilerOptions: {
              sourceMap: true,
            },
          },
        },
      },
    ],
  },
  resolve: {
    extensions: [".tsx", ".ts", ".js"],
  }
};

export default webpackConfig;
