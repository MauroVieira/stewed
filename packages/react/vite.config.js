import { defineConfig } from "vite";
// Node
import { resolve } from "path";
// Plugins
import dts from "vite-plugin-dts";
import peerDepsExternal from "rollup-plugin-peer-deps-external";
import react from "@vitejs/plugin-react";
import { libInjectCss } from "vite-plugin-lib-inject-css";

export default defineConfig({
  plugins: [
    react(),
    libInjectCss(),
    dts({
      copyDtsFiles: true
    }),
  ],
  build: {
    copyPublicDir: false,
    lib: {
      entry: resolve(__dirname, "src/index.ts"),
      formats: ["es"],
    },
    rollupOptions: {
      plugins: [peerDepsExternal()],
      output: {
        assetFileNames: "assets/[name][extname]",
        entryFileNames: "[name].js",
      },
    },
  },
});
