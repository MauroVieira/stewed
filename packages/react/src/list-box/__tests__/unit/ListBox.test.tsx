import React from "react";
// UI Components
import { ListBox } from "../../index";
// Utilities
import { fireEvent, render, act } from "@testing-library/react";

describe("ListBox", () => {
  describe("Snapshots", () => {
    it("should renders default component", () => {
      const { container } = render(
        <ListBox>
          <ListBox.Item>Item</ListBox.Item>
        </ListBox>,
      );

      expect(container).toMatchSnapshot();
    });

    it("should renders group component", () => {
      const { container } = render(
        <ListBox>
          <ListBox.Group title="Group">
            <ListBox.Item>Item</ListBox.Item>
          </ListBox.Group>
        </ListBox>,
      );

      expect(container).toMatchSnapshot();
    });

    it("should render additional classes", () => {
      const { container } = render(
        <ListBox className="other-class">
          <ListBox.Group className="other-group-class">
            <ListBox.Item className="other-item-class">Item</ListBox.Item>
          </ListBox.Group>
        </ListBox>,
      );

      expect(container).toMatchSnapshot();
    });

    it("should render left slot content", () => {
      const { container } = render(
        <ListBox>
          <ListBox.Item leftSlot="Left slot">Item</ListBox.Item>
        </ListBox>,
      );

      expect(container).toMatchSnapshot();
    });

    it("should render right slot content", () => {
      const { container } = render(
        <ListBox>
          <ListBox.Item rightSlot="Right slot">Item</ListBox.Item>
        </ListBox>,
      );

      expect(container).toMatchSnapshot();
    });
  });

  describe("Events", () => {
    it("should trigger onClick event when is clicked", async () => {
      const handleClick = jest.fn();

      const { findByText } = render(
        <ListBox>
          <ListBox.Item onClick={handleClick}>Item</ListBox.Item>
        </ListBox>,
      );

      const item = await findByText("Item");

      act(() => {
        fireEvent.click(item);
      });

      expect(handleClick).toHaveBeenCalledTimes(1);
    });
  });
});
