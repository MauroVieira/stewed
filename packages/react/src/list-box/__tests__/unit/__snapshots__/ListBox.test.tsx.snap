// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListBox Snapshots should render additional classes 1`] = `
<div>
  <div
    class="list-box other-class"
    role="listbox"
  >
    <div
      class="list-box__group other-group-class"
      role="group"
    >
      <div
        class="list-box__item other-item-class"
        role="option"
        tabindex="-1"
      >
        <div
          class="list-box__item__text"
        >
          Item
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ListBox Snapshots should render left slot content 1`] = `
<div>
  <div
    class="list-box"
    role="listbox"
  >
    <div
      class="list-box__item"
      role="option"
      tabindex="-1"
    >
      <div
        class="list-box__item__left"
      >
        Left slot
      </div>
      <div
        class="list-box__item__text"
      >
        Item
      </div>
    </div>
  </div>
</div>
`;

exports[`ListBox Snapshots should render right slot content 1`] = `
<div>
  <div
    class="list-box"
    role="listbox"
  >
    <div
      class="list-box__item"
      role="option"
      tabindex="-1"
    >
      <div
        class="list-box__item__text"
      >
        Item
      </div>
      <div
        class="list-box__item__right"
      >
        Right slot
      </div>
    </div>
  </div>
</div>
`;

exports[`ListBox Snapshots should renders default component 1`] = `
<div>
  <div
    class="list-box"
    role="listbox"
  >
    <div
      class="list-box__item"
      role="option"
      tabindex="-1"
    >
      <div
        class="list-box__item__text"
      >
        Item
      </div>
    </div>
  </div>
</div>
`;

exports[`ListBox Snapshots should renders group component 1`] = `
<div>
  <div
    class="list-box"
    role="listbox"
  >
    <div
      class="list-box__group"
      role="group"
    >
      <div
        class="list-box__group__title"
      >
        Group
      </div>
      <div
        class="list-box__item"
        role="option"
        tabindex="-1"
      >
        <div
          class="list-box__item__text"
        >
          Item
        </div>
      </div>
    </div>
  </div>
</div>
`;
